src/main/java/org/jenkinsci/plugins/codesonar/conditions/NewWarningsIncreasedByPercentageCondition.java

```diff
@@ -58,28 +60,48 @@ public void setWarrantedResult(String warrantedResult) {
     }
 
     @Override
-    public Result validate(CodeSonarBuildActionDTO current, CodeSonarBuildActionDTO previous, Launcher launcher, TaskListener listener, CodeSonarLogger csLogger) {
+    public Result validate(CodeSonarHubAnalysisDataLoader current, CodeSonarHubAnalysisDataLoader previous, String visibilityFilter, String newVisibilityFilter, Launcher launcher, TaskListener listener, CodeSonarLogger csLogger) {
         if (current == null) {
             registerResult(csLogger, CURRENT_BUILD_DATA_NOT_AVAILABLE);
             return Result.SUCCESS;
         }
-        Analysis currentActiveWarnings = current.getAnalysisActiveWarnings();
-        Analysis currentNewWarnings = current.getAnalysisNewWarnings();
+        
+        Long numberOfActiveWarnings = null;
+        try {
+            numberOfActiveWarnings = current.getNumberOfActiveWarnings();
+        } catch (IOException e) {
+            final String applicationMsg = "Error calling number of active warnings on HUB API for current analysis.";
+            LOGGER.log(Level.WARNING, applicationMsg);
+            registerResult(csLogger, applicationMsg);
+            csLogger.writeInfo("Exception: {0}%nStack Trace: {1}", e.getMessage(), Throwables.getStackTraceAsString(e));
+            return Result.FAILURE;
+        }
```

Comment:
At the moment that `IOException `must be catched here since it is a checked exception and there's no `throws `clause on the signature of method `validate()`.
Imagining to add the appropriate `throws `clause to the `validate()` method, the exception can of course float to the class `CodeSonarPublisher` being then shown in the Jenkins' console; there's an important difference in this approach to take into account though.
Such an exception, which originates in the `validate()` method, would compromise the whole Jenkins build with a hard failure.
The consequence will be that other conditions' results, which instead succeeded, will never been reported in the build summary, results will not be persisted and lost.

The following image shows an example of a build summary where only a couple of conditions failed with an exception calling the underlying service:

![image](https://github.com/GrammaTech/jenkins-codesonar-plugin/assets/109066642/186e015e-50a3-489a-94b7-a83a8a93a94b)

This same summary will be like below with the approach of avoiding to catch the exception at condition level:

![image](https://github.com/GrammaTech/jenkins-codesonar-plugin/assets/109066642/cf77938d-86d4-45f0-ae6f-7a6aba6537d5)

Reply:
I am not persuaded that the first screenshot depicts a better solution than the second.  Exceptions should be raised for exceptional situations, and if the plugin fails for an exceptional reason, then I don't think it should bother evaluating any further conditions.  If the build were to fail due to a compilation error, I think it would yield a build result page that looks nearly identical to your second screenshot; the user would still need to look at the build console log to see the nature of the failure.

Reply:
Okay, if that side effect does not represent a problem I will let the exception float!